src/bmtrap.c

```diff
@@ -25,163 +25,158 @@
+    case 0:
+        MU_EndAll();
+        break;
 
-        case 1:
-            MU_EndAll();
-            MU_Create(gActiveUnit);
-            MU_SetDefaultFacing_Auto();
-            break;
+    case 1:
+        MU_EndAll();
+        MU_Create(gActiveUnit);
+        MU_SetDefaultFacing_Auto();
+        break;
 
-        case 2:
-            MU_End(MU_GetByUnit(unit));
-            break;
+    case 2:
+        MU_End(MU_GetByUnit(unit));
+        break;
     }
 
     gActionData.trapType = TRAP_TORCHLIGHT;
     BeginUnitCritDamageAnim(unit, TRAP_TORCHLIGHT);
-
-    return;
 }
 
-void sub_80375A0(struct UnknownBMTrapProc * proc)
+void ApplyTrapDamageReal(struct ProcBmTrap * proc)
 {
-    struct Unit* unit = proc->unit;
+    struct Unit * unit = proc->unit;
 
     ApplyHazardHealing(proc, unit, -10, -1);
 
-    if (GetUnitCurrentHp(unit) == 0) {
-        struct Unit* tmp = gActiveUnit;
+    if (GetUnitCurrentHp(unit) == 0)
+    {
+        struct Unit * tmp = gActiveUnit;
         gActiveUnit = unit;
         
         PidStatsRecordDefeatInfo(unit->pCharacterData->number, 0, 3);
 
-        if (CheckForWaitEvents() != 0) {
+        if (CheckForWaitEvents() != 0)
             RunWaitEvents();
-        }
 
         gActiveUnit = tmp;
     }
-
-    return;
 }
 
 int GetPickTrapType(struct Unit * unit)
 {
     struct Trap * trap;
 
-    if ((trap = GetTrapAt(unit->xPos, unit->yPos)) == NULL) {
-        return 0;
-    }
+    if ((trap = GetTrapAt(unit->xPos, unit->yPos)) == NULL)
+        return TRAP_NONE;
 
     switch (trap->type) {
-        case TRAP_BALLISTA:
-            return 0;
+    case TRAP_BALLISTA:
+        return TRAP_NONE;
 
-        case TRAP_FIRETILE:
-            if ((UNIT_CATTRIBUTES(unit) & CA_THIEF)) {
-                return 0xF;
-            }
-            break;
+    case TRAP_FIRETILE:
+        if ((UNIT_CATTRIBUTES(unit) & CA_THIEF))
+            return TRAP_FIRE_THIEF;
 
-        case TRAP_MINE:
-            if ((UNIT_CATTRIBUTES(unit) & CA_ASSASSIN)) {
-                if (GetUnitItemCount(unit) != 5) {
-                    return 0x10;
-                }
-                return 0;
-            } else if ((UNIT_CATTRIBUTES(unit) & CA_STEAL)) {
-                return 0;
-            }
-            break;
+        break;
+
+    case TRAP_MINE:
+        if ((UNIT_CATTRIBUTES(unit) & CA_ASSASSIN))
+        {
+            if (GetUnitItemCount(unit) != 5)
+                return TRAP_MIN_ASSASSIN;
+
+            return TRAP_NONE;
+        } else if ((UNIT_CATTRIBUTES(unit) & CA_STEAL))
+            return TRAP_NONE;
+
+        break;
     }
 
     return trap->type;
 }
 
 int ExecTrap(ProcPtr proc, struct Unit * unit, int param_3)
 {
-    struct UnknownBMTrapProc* proc2;
+    struct ProcBmTrap * proc2;
 
     switch (GetPickTrapType(unit)) {
         case TRAP_8:
             proc2 = Proc_StartBlocking(sProcScr_ExecTrap8, proc);
-            proc2->unk_50 = param_3;
+            proc2->post_exec_type = param_3;
```

Comment:
Would it be possible to give a more meaningful name to `param_3` based on your review?